tests/ha/single_pod_restart/test_server_pod_restart.py

```diff
@@ -0,0 +1,398 @@
+        self.multipart_obj_path = os.path.join(self.test_dir_path, self.test_file)
+        LOGGER.info("Precondition: Verify cluster is up and running and all pods are online.")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Precondition: Verified cluster is up and running and all pods are online.")
+        LOGGER.info("COMPLETED: Setup operations. ")
+
+    def teardown_method(self):
+        """
+        This function will be invoked after each test function in the module.
+        """
+        if self.s3_clean:
+            LOGGER.info("Cleanup: Cleaning created s3 accounts and buckets.")
+            resp = self.ha_obj.delete_s3_acc_buckets_objects(self.s3_clean)
+            assert_utils.assert_true(resp[0], resp[1])
+        if os.path.exists(self.test_dir_path):
+            system_utils.remove_dirs(self.test_dir_path)
+        LOGGER.info("STARTED: Teardown Operations.")
+        if self.restore_pod:
+            resp = self.ha_obj.restore_pod(pod_obj=self.node_master_list[0],
+                                           restore_method=self.restore_method,
+                                           restore_params={"deployment_name": self.deployment_name,
+                                                           "deployment_backup":
+                                                               self.deployment_backup})
+            LOGGER.debug("Response: %s", resp)
+            assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way")
+            LOGGER.info("Successfully restored pod by %s way", self.restore_method)
+        if self.restore_node:
+            LOGGER.info("Cleanup: Power on the %s down node.", self.node_name)
+            resp = self.ha_obj.host_power_on(host=self.node_name)
+            assert_utils.assert_true(resp, f"Failed to power on {self.node_name}.")
+        if self.restore_ip:
+            LOGGER.info("Cleanup: Get the network interface up for %s ip", self.node_ip)
+            self.new_worker_obj.execute_cmd(cmd=cmd.IP_LINK_CMD.format(self.node_iface, "up"),
+                                            read_lines=True)
+            resp = system_utils.check_ping(host=self.node_ip)
+            assert_utils.assert_true(resp, "Interface is still not up.")
+        LOGGER.info("Cleanup: Check cluster status and start it if not up.")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        if not resp[0]:
+            resp = self.ha_obj.restart_cluster(self.node_master_list[0])
+            assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Removing extra files")
+        for file in self.extra_files:
+            system_utils.remove_file(file)
+        LOGGER.info("Done: Teardown completed.")
+
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-34089")
+    def test_io_server_pod_restart(self):
```

Comment:
Please check if this needs to have three parallel threads for read, write, delete.
If there is already similar test then we can keep this as is.